moped-editor/src/views/projects/projectView/ProjectSummaryEditMap.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from "react";
+import { useMutation } from "@apollo/client";
+import NewProjectMap from "../newProjectView/NewProjectMap";
+import {
+  AppBar,
+  Button,
+  Container,
+  IconButton,
+  Dialog,
+  makeStyles,
+  Toolbar,
+  Typography,
+  Slide,
+  CircularProgress,
+} from "@material-ui/core";
+import { Alert } from "@material-ui/lab";
+import { Close as CloseIcon, Save as SaveIcon } from "@material-ui/icons";
+import { UPDATE_PROJECT_EXTENT } from "../../../queries/project";
+
+const useStyles = makeStyles(theme => ({
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+}));
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+const ProjectSummaryMap = ({
+  projectId,
+  selectedLayerIds,
+  projectExtentGeoJSON,
+  isEditing,
+  setIsEditing,
+  refetchProjectDetails,
+}) => {
+  const classes = useStyles();
+  const [updateProjectExtent, { loading, error }] = useMutation(
+    UPDATE_PROJECT_EXTENT
+  );
+  const [editLayerIds, setEditLayerIds] = useState(selectedLayerIds);
+  const [editFeatureCollection, setEditFeatureCollection] = useState(
+    projectExtentGeoJSON
+  );
+
+  /**
+   * Updates isEditing state to close dialog on cancel button click
+   */
+  const handleClose = () => {
+    setIsEditing(false);
+  };
+
+  /**
+   * Calls update project mutation, refetches data and handles dialog close on success
+   */
+  const handleSave = () => {
+    updateProjectExtent({
+      variables: { projectId, editLayerIds, editFeatureCollection },
+    }).then(() => {
+      refetchProjectDetails();
+      handleClose();
+    });
+  };
+
+  return (
+    <Dialog
+      fullScreen
+      open={isEditing}
+      onClose={handleClose}
+      TransitionComponent={Transition}
```

Comment:
TIL thanks to you! :) Transition objects

Reply:
same